misc/python/materialize/mz_version.py

```diff
@@ -86,6 +86,9 @@ def str_without_prefix(self) -> str:
     def __str__(self) -> str:
         return f"{self.get_prefix()}{self.str_without_prefix()}"
 
+    def is_dev_version(self) -> bool:
+        return str(self).endswith("-dev")
```

Comment:
Is this still correct? At least the endswith suprises me.

Reply:
Yes, I think so. This is `MzVersion` not the image tag. Though, it might be cleaner to access the `prerelease` field:

```
return self.prerelease == "dev"
```

@benesch, do you want to change that? I don't want to get in conflict with your pushes.

Reply:
The prerelease versions have changed to e.g. `v0.114.0-dev.0`, so this logic is no longer correct, and `self.prerelease == "dev"` is also not correct. I'd recommend just checking for the existence of `prerelease` at all:

```python
self.prerelease is not None
```

I'm done pushing for the moment, so feel free to push up a fix for this!

Reply:
Ok, adjusted.

I further fixed the `dev`-suffix removal with https://github.com/MaterializeInc/materialize/pull/29097/commits/d4861c38ffed65b7b866da7bb4b67cf5819d5b69.